### DISCUSSION TITLE
File Based generator with unplugin

### DISCUSSION THREAD

**Discussion by JoSeBu1** [TanStack/router#1141] [Ideas]
The plugin that now is done only for Vite can be write with unplugin and this exports vite, webpack, etc plugins to improve compatibility. 

This is how actually millionjs compiler plugin works.

Thanks!

**Comment by zngly-vlad** [comment]
This is a great idea, would love to see this instead of maintaining plugins for each bundler

+1 https://unplugin.unjs.io/

**Reply by SeanCassiere** [reply]
I've added your comment onto the V2 discussion thread as a [suggestion](https://github.com/TanStack/router/discussions/975#discussioncomment-9003964).

When I have some time, I'll look into trying this out locally and seeing if it would be possible to get the router working with other build-tools.

**Reply by SeanCassiere** [reply]
📢 Got a plan for making the migration to unplugin.

Tracking the migration here: https://github.com/TanStack/router/issues/1756

**Reply by SeanCassiere** [reply]
The router plugin has been moved over to unplugin.
